utils/zerovec/src/ule/mod.rs

```diff
@@ -46,6 +46,16 @@ where
     /// The error that occurs if a byte array is not valid for this ULE.
     type Error;
 
+    /// Validates a byte slice, `&[u8]`.
+    ///
+    /// If `Self` is not well-defined for all possible bit values, the bytes should be validated,
+    /// and `Self::Error` should be returned if they are not valid.
+    ///
+    /// The default implementation validates the slice for all possible bit values.
+    fn validate_byte_slice(_bytes: &[u8]) -> Result<(), Self::Error> {
+        Ok(())
```

Comment:
issue: We should absolutely not have a default impl here given how safety-critical it is. The others are fine having a default because there is an expected invariant, but implementors should not forget to override this.